db/judge/SolidityJudge.js

```diff
@@ -140,6 +163,20 @@ class SolidityJudge extends BaseJudge {
   hash (...args) {
     return this.web3.utils.soliditySha3(...args)
   }
+
+  promisifyTransaction (tx, args) {
```

Comment:
Rename to `awaitTx`?

Reply:
As `promisifyTransaction` "converts" a tx to a promise (https://nodejs.org/api/util.html#util_util_promisify_original) I found it better as a name. `awaitTx` is also fine and smaller as a name. If I change the name of this functions I have to change it everywhere in assignments, so I prefer to keep it as it is.

Reply:
For me the best is `promisifyTX`.

Reply:
The `promisify` name comes from the fact that it converts from *any* callback-style code to promise code, and the caller *needs* to know that the parameter is event-style, but here this is not the intention.

I do not insist on calling it `awaitTx`, but I don't think calling it `promisifyTx` is appropriate. You can call it `waitForConfirmation` or `ensureConfirmed` if you prefer.

The reason is not length, but encapsulation: The user of the API does not need to know that the `tx` offers an event-based API, but can rely on `ensureConfirmed` to deal with the `tx` as it should. The details of implementation of `tx` are encapsulated by `ensureConfirmed`, which allows us to await for it -- whether `tx` offers an event-based API or not. More specifically, if `tx` were ever to offer an await-based API, `ensureConfirmed` could be updated to reflect that in its implementation, and the caller would never know. Such a change should not require a name change to the `ensureConfirmed` function, as it still does the same thing (it returns a promise which ensures a confirmation), albeit differently (by using await instead of creating a promise wrapper around events).

To make the case more concretely, imagine how this would work if it were part of the `tx` object itself: `await tx.ensureConfirmed()` is the correctly encapsulating style. In fact, if we can make this part of the `tx` itself it would be ideal, but I don't think it's necessary to change the web3 API at this point (would it be easy to do so?).

Reply:
As for a global replacement during such refactorings, you can use Atom's Cmd+Shift+F functionality to do a project-wide replacement, or on UNIX use the `sed` command to do a project-wide replacement; don't fix every file by hand!

Reply:
I know man :). Even that is a pain. I have to test it again for each assignment. So `ensureConfirmed` or `awaitTx` ? I don't care. This `await tx.ensureConfirmed()` let's leave it (if it is easy and possible) to another PR